lib/cisco_node_utils/interface.rb

```diff
@@ -72,10 +79,23 @@ def self.interfaces(opt=nil)
       intf_list = config_get('interface', 'all_interfaces')
       return hash if intf_list.nil?
 
+      # Massage intf_list data into an array that is easy
+      # to work with.
+      intf_list.collect! { |x| x.strip || x }
+      intf_list.delete('')
+      intf_list = intf_list.join(' ').split('interface')
+      intf_list.delete('')
```

Comment:
@mikewiebe This massage doesn't work right with the section option. 

```
      intf_list = node.config_get('interface', 'all_interfaces')
# => ["interface Ethernet5/48", "", "  service instance 22 vni", "", ""]

      intf_list.collect! { |x| x.strip || x }
      intf_list.delete('')
      intf_list = intf_list.join(' ').split('interface')
      intf_list.delete('')
      puts intf_list
# => [" Ethernet5/48 service instance 22 vni"]
```

There are also other NU providers that use `all_interfaces` (e.g. `interface_service_vni`) so this list massage needs to occur for all of those providers.

Reply:
I was a little hasty. This code looks okay here but `interface_service_vni` needs similar massage code.